SerialPrograms/Source/PokemonSV/Programs/ItemPrinter/PokemonSV_ItemPrinterRNG.cpp

```diff
@@ -604,6 +618,194 @@ void ItemPrinterRNG::print_again(
     }
 }
 
+void ItemPrinterRNG::run_item_printer_rng_automode(
+    SingleSwitchProgramEnvironment& env, 
+    BotBaseContext& context, 
+    ItemPrinterRNG_Descriptor::Stats& stats
+){
+    const uint16_t min_happiny_dust = 400;
+
+    MaterialFarmerOptions material_farmer_options(
+        true, true,
+        &LANGUAGE,
+        NOTIFICATION_STATUS_UPDATE,
+        NOTIFICATION_PROGRAM_FINISH,
+        NOTIFICATION_ERROR_RECOVERABLE,
+        NOTIFICATION_ERROR_FATAL
+    );
+
+    material_farmer_options.RUN_TIME_IN_MINUTES.set(32);
+    material_farmer_options.SANDWICH_OPTIONS.set_enabled(true);
+    material_farmer_options.SANDWICH_OPTIONS.SAVE_GAME_BEFORE_SANDWICH = true;
+    material_farmer_options.SANDWICH_OPTIONS.BASE_RECIPE.set(BaseRecipe::non_shiny);
+
+    //  For each job that we print, we increment jobs_counter.
+    //  Each time we run the material farmer, we reset jobs_counter to 0.
+    uint32_t jobs_counter = 0;
+
+    // Check material quantity when:
+    // - once when first starting the item printer
+    // - before starting material farming. If still have material, 
+    // can keep using item printer. But this check is only done once, 
+    // until you farm materials again.
+    // - when back from material farming
+    bool done_one_last_material_check_before_mat_farming = false;
+    uint32_t material_farmer_jobs_period = calc_num_jobs_using_happiny_dust(env, context, min_happiny_dust);   
+    bool have_cleared_out_bonus = false;
+    std::map<std::string, uint16_t> obtained_prizes;

```

Comment:
Do you think a map or unordered map would be better here? Or maybe it doesn't matter? The number of items in the map can range anywhere from 5-10 to 120, depending on how many different desired items are selected.

Reply:
If ordering doesn't matter, unordered map are better. But for such small things, it won't matter much in most cases

Reply:
If performance isn't important, I usually prefer map over unordered map because of the consistent sorting. Though the converse can be true if you don't want people to accidentally become dependent on the order.

I think some of the higher level languages (was it Javascript?) or something wanted to randomize the iteration order of hash tables to prevent this specifically. lol